### ISSUE TITLE
[ENH] - Add ability to configure Jhub Apps in nebari config file

### ISSUE THREAD

**Issue by Adam-D-Lewis** [nebari-dev/nebari#2803]
### Feature description

In Nebari Config, add the following section

```yaml
JhubApps:
    - display_name:
      description:
      thumbnail:
      framework:
      filepath:  <insert public git repo>
      conda_environment:
      spawner_profile:
      allow_public_access:
      start_app: False 
      share_with:  [<nebari-usernames-or-groups>]
      user: <insert nebari user>
      retry:  # (optional)
        retry_frequency: '1min'
        max_retries: 10
```

to allow a jhub app to be configured via Nebari config file.  Ideally, this would also retry deployment if the filepath or conda environment aren't present initially e.g. files are still being downloaded, conda env is still being built.  Optionally, allow the app to be created in Jhub apps but don't start it yet.  Then leave the inputs unvalidated until the user goes and starts the app via the GUI.

### Value and/or benefit

This would enable the creation of a RAG distribution of Nebari.  The goal would be to allow people to easily try out Ragna and other AI apps in the future.  This would be done by creating a `nebari-ragna` plugin which adds a --distr=ragna cli arg to nebari init which would configure a ragna conda env, gpus, and sets up the ragna app via jhub apps by adding to the nebari config file.

See https://docs.google.com/document/d/1mlZHAxIusTdYlnT6rfHxCpXi88fND2Cs1EDzgiL1Gxc for more details.

### Anything else?

I'm not sure of tho following:
- Can jhub apps automatically run from a git repo?
- Can jhub apps input validation (e.g.filepath and conda env exist) be postponed until user tries to start the app?
- My understanding is that all Jhub apps need to be created by a jhub user.  We may need to create a Nebari user for these apps to belong to?

**Comment by krassowski** [comment]
Should jhub-apps overrides config be used for it or should it be a separate section? For reference:
- https://github.com/nebari-dev/nebari/issues/2753
- https://github.com/nebari-dev/nebari/pull/2754

Also looks like there is some overlap with the work for deploying from a git repo which has a config file

**Comment by Adam-D-Lewis** [comment]
> Should jhub-apps overrides config be used for it or should it be a separate section?

I think it belongs in a separate section from overrides.

> Also looks like there is some overlap with the work for deploying from a git repo which has a config file

Could you link to that work here @krassowski so I can look into it further?

**Comment by Adam-D-Lewis** [comment]
It looks like https://github.com/nebari-dev/jhub-apps/blob/023fc57edf62411600d39ae1a607f9a13027a718/jhub_apps/tests/tests_e2e/test_api.py#L21 and https://github.com/nebari-dev/jhub-apps-from-git-repo-example/ could be similar/useful to this feature.

**Comment by Adam-D-Lewis** [comment]
One thing I don't understand is where is the app config data stored?  Does it end up in the jupyterhub database somehow?

**Comment by Adam-D-Lewis** [comment]
Okay, more details on this.  We should be able to specify which user the app belongs to.  Should be able to delete other servers of that user that are not passed in.
